### DISCUSSION TITLE
DIAMOND v2.1.0

### DISCUSSION THREAD

**Discussion by bbuchfink** [bbuchfink/diamond#646] [General]
- Added the `cluster` workflow to cluster protein sequences.
- Added the `realign` workflow to generate clustering output.
- Added the `recluster` workflow to correct errors in clusterings.
- Added the `reassign` workflow to reassign cluster members to their closest centroid.
- Added the option `-M/--memory-limit` to set a memory limit for clustering workflows.
- Added the `approx-id` option to filter alignments by approximate sequence identity and to set an approximate sequence identity threshold for clustering.
- Added the `--member-cover` option to set the coverage threshold of the cluster member sequence.
- Added the `cluster-steps` option to set steps for cascaded clustering.
- Added the `clusters` option to specify clustering input file.
- The `blastx` mode will now mask any open reading frame below the minimum required length as specified by `--min-orf`.
- The `blastx` mode will only count unmasked letters towards the block size.
- Fixed a bug that caused an error when using the global ranking mode.
- Added the fast mode as the first round in iterative searches.
- Fixed a bug that caused the program not to function on systems without support for SSE4.1.
- Improved multi-threaded load balancing of gapped extension computations.
- Improved performance of seed extension stage when HSP filter settings are used.
- Added the option `--soft-masking` with possible values `0` and `tantan` to permit soft-masking using the tantan algorithm.
- Fixed a bug that could cause an `inflate error` in multiprocessing mode.
- Added the option `--swipe` to compute full Smith Waterman alignments of all queries against all targets.
- Added the sensitivity mode `--faster`.
- Added the option `-M` to set memory limit for the `cluster`, `realign`, `recluster` and `reassign` workflows.
- Added the output fields `approx_pident` and `corrected_bitscore` to the tabular format.
- Added the `--lin-stage1` option to linearize comparisons in the seeding stage by only considering hits against the longest query sequence for identical seeds.
- Added the `--kmer-ranking` option to rank sequences when `--lin-stage1` is used (only supported when compiled with `-DKEEP_TARGET_ID=ON`).
- Added the option `--no-block-size-limit` to deactivate upper limits for the block size when the `--memory-limit` option is used.
- Added the `greedy-vertex-cover` workflow to compute clustering based on alignments.
- Added the `--edge-format` option to set edge format for greedy vertex cover.
- Added the `--edges` option to set input file for greedy vertex cover.
- Added the `--centroid-out` option to output centroid sequences for greedy vertex cover.
- Added the `--unaligned-targets` option to generate an output file of unaligned targets.
- Fixed an issue that failed compilation using the Intel Compiler.
- Fixed an issue that could cause a segmentation fault in rare cases.
- The `--header` option can now be used with the parameter `simple` to enable simple headers for the tabular format, or without a parameter to enable headers for the clustering format.
- Added the option `--mp-self` to optimize self-alignment in multiprocessing mode.
- Added the option `--query-or-subject-cover` to report alignments if the query or the subject cover (or both) are above the given threshold.
- Removed support for the `--comp-based-stats 2` option (now equivalent to `--comp-based-stats 3`).
- Removed hit culling in case of overlapping target ranges in frameshift alignment mode.
- Added the option `--anchored-swipe` to activate anchored SWIPE extension.

<hr /><em>This discussion was created from the release <a href='https://github.com/bbuchfink/diamond/releases/tag/v2.1.0'>DIAMOND v2.1.0</a>.</em>

**Comment by taylorreiter** [comment]
Would you be willing to provide an explanation for the `corrected_bitscore`? I tried to search for documentation on what type of correction this is, but I couldn't find it. I was very excited to see this addition though!

**Reply by bbuchfink** [reply]
It's basically the correction described here: https://bmcresnotes.biomedcentral.com/articles/10.1186/1756-0500-5-286

I adapted this for correcting the bit score instead of the e-value (based on the output of the ALP library).

**Reply by taylorreiter** [reply]
Thank you so much for your response, this is super helpful to see!

**Comment by V-Varga** [comment]
I apologize for the trouble, but could you elaborate on how you calculate percent identity for `--approx-id`? I have searched the documentation, but the best I can find is "the approximate sequence identity derived as a linear regression from the bitscore, not the actual number of identities in the alignment" from the Wiki page for Clustering (https://github.com/bbuchfink/diamond/wiki/Clustering). This sounds distinct from the BLAST definition (as I understand it: the number of identical residues/amino acids divided by the number of alignment columns).

**Reply by bbuchfink** [reply]
The formula to compute it is given in the supplementary information of our clustering preprint.

**Reply by V-Varga** [reply]
Thank you very much for taking the time to respond! This was very helpful, exactly what I was looking for!
